modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/AbstractPageMemoryMvPartitionStorage.java

```diff
@@ -50,51 +49,44 @@ public abstract class AbstractPageMemoryMvPartitionStorage implements MvPartitio
     private static final Predicate<BinaryRow> MATCH_ALL = row -> true;
 
     private static final Predicate<Timestamp> ALWAYS_LOAD_VALUE = timestamp -> true;
-    private static final Predicate<Timestamp> NEVER_LOAD_VALUE = timestamp -> false;
-    private static final Predicate<Timestamp> LOAD_VALUE_WHEN_UNCOMMITTED = RowVersion::isUncommitted;
 
-    private final int partId;
+    private final int partitionId;
     private final int groupId;
 
-    private final VersionChainFreeList versionChainFreeList;
     private final VersionChainTree versionChainTree;
-    private final VersionChainDataPageReader versionChainDataPageReader;
     private final RowVersionFreeList rowVersionFreeList;
     private final DataPageReader rowVersionDataPageReader;
 
-    private final ThreadLocal<ReadRowVersion> readRowVersionCache = ThreadLocal.withInitial(ReadRowVersion::new);
-    private final ThreadLocal<ScanVersionChainByTimestamp> scanVersionChainByTimestampCache = ThreadLocal.withInitial(
-            ScanVersionChainByTimestamp::new
-    );
+    private final ThreadLocal<ReadRowVersion> readRowVersionCache;
```

Comment:
Does this class really make sense? `ReadRowVersion` is a minuscule class with a minimum amount of data. Are you sure it would be faster to bother with `ThreadLocal` instead of creating a new instance?

Reply:
No, I'm not sure about it. Allocation is fast enough. Should I get rid of these?